hw/top_earlgrey/dv/chip_sim_cfg.hjson

```diff
@@ -295,9 +295,9 @@
       en_run_modes: ["sw_test_mode"]
     }
     {
-      name: chip_sw_aon_timer_wakeup_irq
+      name: chip_sw_aon_timer_irq
```

Comment:
Could you name this test `chip_sw_aon_timer_wdog_bark_irq` for more clarity? And likewise for the C test.

Reply:
Nevermind - I now realize you have combined testing of both irqs in the same test.

Reply:
Please map this test in the testplan to the wdog bark irq testpoint so that it shows up as covered.

Reply:
The test is checking the wakeup and wdog bark irqs, so why should I call it `chip_sw_aon_timer_wdog_bark_irq`?
Maybe we should give a name that represents both tests.

Reply:
@sriyerg do you think I should increase the `sw_test_timeout_ns`?

Reply:
It would be great if you could run this in DV and adjust accordingly.

Reply:
@sriyerg the test took ~17min to run locally.
```
                [   legend  ]: [Q: queued, D: dispatched, P: passed, F: failed, K: killed, T: total]                                      
00:01:15  [    build    ]: [Q: 0, D: 0, P: 1, F: 0, K: 0, T: 1] 100%                                                                  
00:16:42  [     run     ]: [Q: 0, D: 0, P: 1, F: 0, K: 0, T: 1] 100%   
```
What would be the appropriate value for the `sw_test_timeout_ns`? What is the logic behind it?

Reply:
How long we simulate the chip (typically, upto few milliseconds), aka simtime impacts the runtime (wall clock time). It typically takes under 5 minutes to simulate 4ms on the chip. The `sw_test_timeout_ns` (which is a simtime timeout) is thus, set to 4ms by default. If a specific test needs to go over, then it needs to increase this timeout accordingly. If the timeout is exceeded, the test is killed, and the test is considered to have failed. This is a general timeout that applies to ALL tests.    

If you open the `run.log`, which will be in the run output directory (`scratch/<branch>/top_earlgre-sim-vcs/0.test-name/out/run.log`). The last UVM_INFO message will typically provide the simtime for this test. That is how long it took for this particular randomly picked seed for this test to run. Different seeds may incur different runtimes depending on what is randomized in the test. For the timer / wdog tests, we randomize the length of the timeout. If you run the limit of what you are randomizing (in this case, 15ms timer timeout), you get the worst case simtime. Setting that (+ 20% buffer) that as the simtime timeout should be appropriate.  

I probably agree with @matutem here - we could reduce the AON wake up time even further to reduce the runtime. But we will run this test mainly in the nightly regressions, so the ~15 minute runtime is not as worrisome to me.

Reply:
I see, I'll run some tests to check how the decrease of the timeout would affect the simulation time.

Reply:
I managed to reduce the execution time to 8 min in the worst scenario (1500 us). Bear in mind we are executing 2 tests here (`chip_sw_aon_timer_wdog_bark_irq` and `chip_sw_aon_timer_wakeup_irq`) which means 4 minutes for each.
@sriyerg and @matutem do you think that is enough?

Reply:
I am fine with upto 30 minute runtime (if the test really warrants it), but everyone has different tolerance level. For this test in particular, <10 min is reasonable. There really is no need to test large timeouts for the AON timer or the WDOG. The block is well verified at its level.